cns/logger/v2/logger.go

```diff
@@ -0,0 +1,44 @@
+package logger
+
+import (
+	cores "github.com/Azure/azure-container-networking/cns/logger/v2/cores"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type Config struct {
+	// Level is the general logging Level. If cores have more specific config it will override this.
+	Level      zapcore.Level
+	AIConfig   cores.AIConfig
+	FileConfig cores.FileConfig
+}
+
+type compoundCloser []func()
+
+func (c compoundCloser) Close() {
+	for _, closer := range c {
+		closer()
+	}
+}
+
+func New(cfg *Config) (*zap.Logger, func(), error) {
+	stdoutCore := cores.StdoutCore(cfg.Level)
+	closer := compoundCloser{}
+	fileCore, fileCloser, err := cores.FileCore(&cfg.FileConfig)
+	closer = append(closer, fileCloser)
+	if err != nil {
+		return nil, closer.Close, err //nolint:wrapcheck // it's an internal pkg
+	}
+	aiCore, aiCloser, err := cores.ApplicationInsightsCore(&cfg.AIConfig)
+	closer = append(closer, aiCloser)
+	if err != nil {
+		return nil, closer.Close, err //nolint:wrapcheck // it's an internal pkg
+	}
+	platformCore, platformCloser, err := platformCore(cfg)
+	closer = append(closer, platformCloser)
+	if err != nil {
+		return nil, closer.Close, err
+	}
+	core := zapcore.NewTee(stdoutCore, fileCore, aiCore, platformCore)
```

Comment:
It might be nice to have this be flexible instead of requiring all three.

Reply:
I'm trying to balance between it being customizable, and it being opinionated, since it is the core CNS logger and I want to to generally behave in a consistent way. Are there scenarios where this should be variable?

Reply:
Tried this. I think I even managed to get it to where knowledge of the platform specific bits doesn't escape this package (even to the global Config struct). The tradeoff on that is that there is no defaulting on the config - it will run only exactly what exactly how tell it in the config file, and zero values are legal.

Reply:
@rbtr awesome :) To answer your earlier question, Jackie and I are currently working on a proposal to rip AI out, so the sink config may change. The flexibility is appreciated.